### PR TITLE
Multi histogram creation with polarization options

### PR DESCRIPTION
**Pull request by mpatrou**
The code implements the creation of multiple histograms with polarization options SF and NSF. It includes:

- MakeMultipleSlices Mantid Algorithm
- Related functions in Histogram creation tab are updated to work for single- and multi- histogram creation.
- Validation and error/warning message workflow update on "Histogram Submit" button. Users see the warning, when the flipping ratio sample logs stored in SF and NSF workspaces are different or the flipping ratio is stored only in SF. Users can select to continue or cancel the workflow.
- MakeSlice for any single workspace with any polarization state is enabled
- One Background workspace for multi- histogram creation is enabled
- ImportErrors when running tests with MakeSlice/MakeMultipleSlices one-by-one fixed. Shiver was imported on top of the file.
- github actions updated for tests to pass. 
- Tests for the whole workflow and MakeMultipleSlices algorithm added; including test files

### CODE REVIEW

File: src/shiver/models/makeslices.py [hunk 1]
```diff
@@ -0,0 +1,239 @@
...
+            IMDHistoWorkspaceProperty(
+                "NSFOutputWorkspace", defaultValue="", optional=PropertyMode.Mandatory, direction=Direction.Output
+            ),
+            doc="Non Spin-Flip OutputWorkspace IMDHisto workspace",
+        )
+
+    def PyExec(self):
+        flipping_ratio = self.getPropertyValue("FlippingRatio")
+
+        if flipping_ratio == "":
```
Comment (AndreiSavici) [inline src/shiver/models/makeslices.py:105]: Add a validator instead in the PyInit function

File: src/shiver/models/makeslices.py [hunk 2]
```diff
@@ -0,0 +1,239 @@
...
+    MakeSlice,
+    FlippingRatioCorrectionMD,
+    _create_algorithm_function,
+)
+
+
+from shiver.version import __version__
+
+
+class MakeMultipleSlices(DataProcessorAlgorithm):
```
Comment (AndreiSavici) [inline src/shiver/models/makeslices.py:31]: Rename it to `MakeSFCorrectedSlices`

File: src/shiver/models/makeslices.py [hunk 3]
```diff
@@ -0,0 +1,239 @@
...
+        self.declareProperty(
+            name="FlippingRatio",
+            defaultValue="",
+            direction=Direction.Input,
+            doc="""Formula to define the flipping ratio. It can depend on the variables in the list of sample logs
+          defined below""",
+        )
+
+        self.declareProperty(
+            name="FlippingSampleLog",
```
Comment (AndreiSavici) [inline src/shiver/models/makeslices.py:61]: name should be `FlippingRatioSampleLog`

**Review by AndreiSavici — CHANGES_REQUESTED** [review]
In addition to the minor changes, I've found a small usability bug as well. I've started Shiver from within workbench, I've loaded the files, selected them as SF data and NSF data. Then I add the logs. The workspace type change from D_N to Q_S. Then when I click on the workspace, I can't set it to NSF, only to unpolarized or SF. Similar to the other workspace

File: src/shiver/models/histogram.py [hunk 4]
```diff
@@ -293,15 +294,26 @@ def save_polarization_state(self, name, pol_state):
     def get_polarization_state(self, name):
         """Get the polarization state from Sample Log in workspace"""
         workspace = mtd[name]
-        pol_state = "UP"
+        pol_state = self.get_experiment_sample_log(workspace, "polarization_state")
+        if pol_state is None:
+            # revert to default unpolarized state
+            pol_state = "UP"
```
Comment (AndreiSavici) [inline src/shiver/models/histogram.py:300]: Please also change "UP" to "UNP"

File: src/shiver/models/makeslices.py (hunk 1 above)
Comment (mpatrou) [reply src/shiver/models/makeslices.py]: done!

File: src/shiver/models/makeslices.py (hunk 2 above)
Comment (mpatrou) [reply src/shiver/models/makeslices.py]: done

File: src/shiver/models/makeslices.py (hunk 3 above)
Comment (mpatrou) [reply src/shiver/models/makeslices.py]: done

File: src/shiver/models/histogram.py (hunk 4 above)
Comment (mpatrou) [reply src/shiver/models/histogram.py]: done

**Review by AndreiSavici — APPROVED** [review]

**Pull request merged**